cmd/jaeger/internal/extension/jaegerstorage/extension_test.go

```diff
@@ -129,6 +134,67 @@ func TestStorageExtension(t *testing.T) {
 	require.NoError(t, err)
 }
 
+func TestBadgerStorageExtension(t *testing.T) {
+	ctx := context.Background()
+	telemetrySettings := component.TelemetrySettings{
+		Logger:         zap.NewNop(),
+		TracerProvider: nooptrace.NewTracerProvider(),
+		MeterProvider:  noopmetric.NewMeterProvider(),
+	}
+
+	tempDir := t.TempDir()
+	defer os.RemoveAll(tempDir)
+
+	config := &Config{
+		Badger: map[string]badgerCfg.NamespaceConfig{
+			badgerName: {
+				ValueDirectory:        tempDir,
+				KeyDirectory:          tempDir,
+				Ephemeral:             true,
+				MaintenanceInterval:   5,
+				MetricsUpdateInterval: 10,
+			},
+		},
+	}
+
+	require.NoError(t, config.Validate())
+
+	extensionFactory := NewFactory()
+	storageExtension, err := extensionFactory.CreateExtension(ctx, extension.CreateSettings{
+		ID:                ID,
+		TelemetrySettings: telemetrySettings,
+		BuildInfo:         component.NewDefaultBuildInfo(),
+	}, config)
+	require.NoError(t, err)
+
+	host := componenttest.NewNopHost()
+	err = storageExtension.Start(ctx, host)
+	require.NoError(t, err)
+
+	err = storageExtension.Start(ctx, host)
+	require.Error(t, err)
+	require.EqualError(t, err, fmt.Sprintf("duplicate badger storage name %s", badgerName))
+
+	t.Cleanup(func() { require.NoError(t, storageExtension.Shutdown(ctx)) })
```

Comment:
the point of cleanup function is to declare it early so that it's actually guaranteed to be called. If require above raises an error this line won't be reached

Reply:
moved it above `require`